components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -203,63 +177,71 @@ public Response updateResource(String resourceId, ResourceDetailsDTO updatedReso
      * @param resourceId resourceId of the resource which need to get deleted
      * @return Response with the status of resource deletion
      */
-    @Override
-    public Response deleteResource(String resourceId, MessageContext messageContext) {
 
-        Response response = null;
+    @Override
+    public Response deleteResource(String resourceId, MessageContext context) {
 
         try {
-            if (isResourceId(resourceId)) {
-
-                try {
-                    if (ResourceUtils.getResourceService().deleteResource(resourceId)) {
-                        return Response.status(Response.Status.NO_CONTENT).build();
-                    }
-
-                } catch (UMAServiceException e) {
-                    handleErrorResponse(e, log);
-                    log.error("Invalid request. ", e);
-
-                } catch (Throwable throwable) {
-                    handleErrorResponse((UMAException) throwable, log);
-                    log.error("Internal server error occurred. ", throwable);
+            if (isValidToken(context) && isResourceIdValid(resourceId)) {
+                if (ResourceUtils.getResourceService().deleteResource(resourceId)) {
+                    return Response.status(Response.Status.NO_CONTENT).build();
                 }
-            } else {
-                throw new UMAClientException(ResourceConstants.ErrorMessages.ERROR_CODE_NOT_FOUND_RESOURCE_ID);
             }
+        } catch (UMAServiceException e) {
+            log.error("Invalid request.", e);
```

Comment:
Any specific reason for logging errors or stacktrace on invalid requests ? Ideally we don't do this.